azure-mgmt-monitor/src/test/java/com/microsoft/azure/management/redis/MonitorActivityAndMetricsTests.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.management.redis;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.management.compute.VirtualMachine;
+import com.microsoft.azure.management.monitor.EventData;
+import com.microsoft.azure.management.monitor.MetricCollection;
+import com.microsoft.azure.management.monitor.MetricDefinition;
+import com.microsoft.azure.management.monitor.ResultType;
+import org.joda.time.DateTime;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.List;
+
+public class MonitorActivityAndMetricsTests extends MonitorManagementTest {
+    @Test
+    public void canListEventsAndMetrics() throws Exception {
+        DateTime recordDateTime = DateTime.parse("2018-01-24T00:07:40.350Z");
+        VirtualMachine vm = computeManager.virtualMachines().list().get(0);
+
+        // Metric Definition
+        List<MetricDefinition> mt = monitorManager.metricDefinitions().listByResource(vm.id());
+
+        Assert.assertNotNull(mt);
+
+        // Metric
+        MetricCollection metrics = mt.get(0).defineQuery()
+                .withStartTime(recordDateTime.minusDays(30))
```

Comment:
Probably a more fluent way to express the same: "startingFrom", "endsBefore"/"endsAt" depending whether the end time is included in the query or not; "Time" can be omitted since it's implicit given the type of the argument to be specified.

Reply:
fixed.